user_topics: Validate 'topic' parameter length <= max_topic_length.

prakhar1144 opened the pull request:
For endpoints:
* [`/users/me/subscriptions/muted_topics` ](https://zulip.com/api/mute-topic)
* [`/user_topics`](https://zulip.com/api/update-user-topic)

`topic` length was not validated before DB operations which resulted in an exception.

This commit adds a validation for the topic name length <= 'max_topic_length' characters.

Fixes #27796.


<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

<details><summary>Docs</summary>
<img src="https://github.com/zulip/zulip/assets/56781761/68742b23-1ad0-4cbe-8d8f-02cc277328dd">
</img>
</details> 

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

prakhar1144 commented:
[Posted a question in CZO](https://chat.zulip.org/#narrow/stream/378-api-design/topic/max.20topic.20length.20limit.20for.20.2Fuser_topics/near/1712322)

timabbott commented on zerver/views/user_topics.py:
```diff
@@ -66,10 +67,11 @@ def update_muted_topic(
     stream_id: Optional[int] = REQ(json_validator=check_int, default=None),
     stream: Optional[str] = REQ(default=None),
-    topic: str = REQ(),
+    topic: str = REQ(converter=lambda var_name, data: data.strip()),
```
How did you pick this `converter`? We don't use those often, and I'm reluctant to add a new one without tests, since I have a vague recollection of them having slightly different JSON decoding behavior?

prakhar1144 replied:
We use this converter in send-message codepath (see REQ_topic). Since we no longer plan to truncate & client should take care of such things; this is not needed and we can simply validate the topic length.

prakhar1144 commented:
Updated the PR as per our [discussion in CZO](https://chat.zulip.org/#narrow/stream/378-api-design/topic/max.20topic.20length.20limit.20for.20.2Fuser_topics/near/1712672)

Added tests.

timabbott commented:
Merged, thanks @prakhar1144!
